recipe_scrapers/_utils.py

```diff
@@ -93,10 +93,11 @@ def url_path_to_dict(path):
     )
     regex = re.compile(pattern)
     matches = regex.match(path)
-    url_dict = matches.groupdict() if matches is not None else None
+    url_dict = matches.groupdict() if matches else None
 
     return url_dict
 
 
-def get_host_name(url):
-    return url_path_to_dict(url.replace("://www.", "://"))["host"]
+def get_host_name(url: str) -> str:
+    data = url_path_to_dict(url.replace("://www.", "://"))
+    return data.get("host", "UNKNOWN HOST") if data else "UNKNOWN HOST"
```

Comment:
What's the goal of the unknown host fallback case here?

Reply:
`url_path_to_dict` has a chance to return `None` due to the regex function. It's probably unlikely for this to be a problem since this is part of the generator code, but I wanted to handle it. I can remove it if need be.